internal/utils/utils.go

```diff
@@ -235,11 +235,18 @@ func IsPrivateIP(ip net.IP) bool {
 	return false
 }
 
+type CXKey struct {
+	shardID  uint32
+	blockNum uint64
+}
+
 // GetPendingCXKey creates pending CXReceiptsProof key given shardID and blockNum
 // it is to avoid adding duplicated CXReceiptsProof from the same source shard
-func GetPendingCXKey(shardID uint32, blockNum uint64) string {
-	key := strconv.FormatUint(uint64(shardID), 10) + "-" + strconv.FormatUint(blockNum, 10)
-	return key
+func GetPendingCXKey(shardID uint32, blockNum uint64) CXKey {
```

Comment:
can the difference of key format returned cause any issue in our codebase calling that function ?

Reply:
Cache keeped only in the memory, so, restart will erase everything. 
Exactly in this case, we have 3 memory allocations for no reason. My version has no allocations.